### ISSUE TITLE
`IDE0301` gets reported for `Array.Empty<T>()` inside Ternary Expression (and possibly other expressions?)

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn#70903]
**Version Used**: 
Visual Studio Version `17.8.0`
**Steps to Reproduce**:

1. new empty project
2. use e.g. this code:
```cs
using System;
using System.Collections.Generic;
using System.Diagnostics;

namespace DesignLibrary
{
  public sealed class Test1
  {
    private string[] Parse(string raw) => throw new NotImplementedException();

    public void Do(string raw)
    {
      IEnumerable<string> lines = string.IsNullOrEmpty(raw) ? Array.Empty<string>() : Parse(raw);
    }
  }
}
```
![image](https://github.com/dotnet/roslyn/assets/71026023/1b1aae12-4df2-43ba-bd1f-686944021388)

This also happens, wenn `lines` is of type `IReadOnlyCollection<string>` or `ICollection<string>`.

**Diagnostic Id**:

IDE0301

**Expected Behavior**:
`Array.Empty<T>()` should not be reported: it's very likely there for a reason.

**Actual Behavior**:
`Array.Empty<T>()` gets reported, when inside ternary.

**My Reasoning**:
This "silently" goes against the `CA1825: Avoid zero-length array allocations` rule.
The new `[]` initializer is nice and short, however, it will result in a `new List<T>()`, which may result in unused List-Instances, when there is a known case, where Resulting List/Collection/Enumerable is never "written"/Added to.
A lot of our code also does not use `IReadOnlyCollection<T>` or `IReadOnlyList<T>` in the first place, due to age.

Also: `IDE0301` doesn't get reported, when assigning `Array.Empty<T>()` directly, returning it directly, and using a null-coalesce:
![image](https://github.com/dotnet/roslyn/assets/71026023/b1e1e2e6-d926-42c0-8832-cb1a16dc2625)
![image](https://github.com/dotnet/roslyn/assets/71026023/ece0cf44-4f51-44a9-8bdf-3c979f2977f2)

**Comment by CyrusNajmabadi** [comment]
> The new [] initializer is nice and short, however, it will result in a new List<T>()

This is incorrect. It will result in the empty array Singleton.

> This "silently" goes against the CA1825: Avoid zero-length array allocations rule.

This is incorrect. There is no allocation here. It is equivalent to using Array.Empty here
